### ISSUE TITLE
Disable input handling in a non-tty environment

### ISSUE THREAD

**Issue by thetoxicavenger** [vadimdemedes/ink#40]
`/node_modules/ink/index.js:61`

Never seen this one before and googling didn't help much - what's going on here? It's preventing me from running this example from your docs:

`const {h, render, Component} = require('ink');

class Counter extends Component {
	constructor() {
		super();

		this.state = {
			i: 0
		};
	}

	render(props, state) {
		return `Iteration #${state.i}`;
	}

	componentDidMount() {
		this.timer = setInterval(() => {
			this.setState({
				i: this.state.i + 1
			});
		}, 100);
	}

	componentWillUnmount() {
		clearInterval(this.timer);
	}
}

const unmount = render(<Counter/>);

setTimeout(() => {
	// Enough counting
	unmount();
}, 1000);`

**Comment by vadimdemedes** [comment]
What Node.js version are you on?

**Comment by thetoxicavenger** [comment]
8.2.1

**Comment by TheSuperFly** [comment]
This happened when I tried to launch my Ink app on WebStorm IDE, even more while using `node inspect index.js`

I think this error happen when you're trying to run a ink app in a non-tty env such debuggers or some IDE terminal.

**Comment by vadimdemedes** [comment]
Yes, I think it happens only in non-tty env, which I guess there's no way to enable interactive input. Possible solution for this would be to just disable the input altogether.

**Comment by TheSuperFly** [comment]
I've just been looking at it and making a conditional to see if current env is TTY. 

```
readline.emitKeypressEvents(stdin);

if (stdout.isTTY) { // < this
  stdin.setRawMode(true);
}
```

And it worked, it's now working on both debugger, WebStorm IDE and regular terminal (iTerm). Should I make a PR for this fix?

**Comment by vadimdemedes** [comment]
Sure, a PR would be great, thanks! I'd completely disable all handling of stdin, not just `setRawMode()`, but also in `render()` and `exit()` in `index.js`.

**Comment by vadimdemedes** [comment]
Fixed via #42. Will close the issue when a release is out.

**Comment by sonaye** [comment]
Can we have this released please?

**Comment by vadimdemedes** [comment]
Released `0.4.0`, sorry for the delay.
